packages/common/src/router-v2.js

```diff
@@ -0,0 +1,86 @@
+const { getRoutePaths } = require('./router');
+const { Router } = require('express');
+
+/**
+ * @typedef {import('express').IRouter} IRouter
+ * @typedef {import('express').Handler} Handler
+ * @typedef {Object<string, IRouter>} RouteDict
+ * @typedef {import('./router-v2-types').HttpMethods} HttpMethods
+ * @typedef {import('./router-v2-types').RouterModule} RouterModule
+ * @typedef {{ includeRoot?: boolean, backwardsCompatibilityModeEnabled?: boolean }} Options
+ */
+
+/** @type {Array<HttpMethods>} */
+const HttpMethods = [
+	'connect',
+	'delete',
+	'get',
+	'head',
+	'options',
+	'patch',
+	'post',
+	'put',
+	'trace'
+];
+
+/** @type {(str: *) => str is HttpMethods} */
+const stringIsRecognisedExport = (str) => HttpMethods.includes(str);
+
+/**
+ * @param {string} [directory]
+ * @param {Options} [options]
+ * @returns {IRouter}
+ */
+exports.getRouter = (
+	directory = __dirname,
+	{ backwardsCompatibilityModeEnabled = false, ...options } = {
+		includeRoot: false,
+		backwardsCompatibilityModeEnabled: false
+	}
+) =>
+	getRoutePaths(directory, options)
+		.sort((a, b) => b.split('/').length - a.split('/').length)
```

Comment:
nice to have a comment explaining why we're sorting e.g.
// sorting by folder depth to ensure most specific routes are applied first to avoid any catch all routes further down